### PR TITLE
Use Splitters in the HttpUriBuilder

### PR DESCRIPTION
**Pull request by wendigo**
# Airlift contribution check list

 - [x] Git commit messages follow https://cbea.ms/git-commit/.
 - [x] All automated tests are passing.
 - [x] Pull request was categorized using one of the existing labels.

### CODE REVIEW

**Comment by wendigo** [comment]
@findepi ptal

**Review by electrum — APPROVED** [review]

File: http-client/src/main/java/io/airlift/http/client/HttpUriBuilder.java [hunk 1]
```diff
@@ -244,13 +248,11 @@ private ListMultimap<String, String> parseParams(String query)
...
-                    .omitEmptyStrings()
-                    .split(query);
-
-            for (String pair : pairs) {
-                String[] parts = pair.split("=", 2);
-                result.put(percentDecode(parts[0]), percentDecode(parts[1]));
+            for (String pair : QUERY_SPLITTER.split(query)) {
+                List<String> parts = QUERY_PAIR_SPLITTER.splitToList(pair);
+                String key = percentDecode(parts.get(0));
+                String value = parts.size() > 1 ? percentDecode(parts.get(1)) : null;
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/HttpUriBuilder.java:254]: I think the old code would fail if there was value, which means we don't have any tests for this case. Add a test to verify that using null here is safe and won't break elsewhere.
